src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java

```diff
@@ -435,6 +435,13 @@ public void testParseDoubleIntegerValues() {
             .isEqualTo(59.0);
     }
 
+    @Test
+    public void testParseDoubleIntegerValues2() {
```

Comment:
Why do we need this test?

Reply:
I added this test because after removing this test https://github.com/checkstyle/checkstyle/blob/74dc9efbc5cf1754dd924a43e059bb4679283a6a/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java#L209-L211 this condition got mutated 
![image](https://github.com/checkstyle/checkstyle/assets/97679350/b6e8ea90-c80c-4bc6-b10d-ed2d401935a8)
To kill this I have added this extra test

Reply:
`parseNumber` is called for `NUM_INT` and `NUMB_LONG` only. I don't see how `txt`can be empty. If you can find an input test which triggers that condition, let's keep it, otherwise, remove it and regenerate reports.

Reply:
You are right, I also not find any way to that txt become 0